app/controllers/papers_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class PapersController < ApplicationController
+  # /semesters/1/paper - lists papers
+  def index
+    return unless ensure_params_fields([:semester_id])
```

Comment:
You don't need this -- by default without semester_id, rails won't even direct the request.